### ISSUE TITLE
Version 1.5.2-alpha does not load any exissting contacts

### ISSUE THREAD

**Issue by arkascha** [owncloud/contacts#552]
I am using the contacts app in version 1.4.0.0 without any issues on an owncloud-9.1.2 system. 

After upgrading to app version 1.5.1 the loading of addresses stopped working. The app itself loads, initializes, the existing address books are recognized, but the spinner indicating the loading of the entries themselves spins forever without producing any results. There are 3 address book inside owncloud holding 332 contacts in total.
@DeepDiver1975 provided a fix in PR https://github.com/owncloud/contacts/pull/551 as version 1.5.2-alpha which I currently test. It shown described behavior for me. 

I made sure the new app is loaded (I did not receive any update notification of owncloud after replacing the apps code...) by removing the js asset file from the pipeline. That indeed changed what I see in the browsers JS console. Version 1.5.1 showed the error described in mentioned pull request, that apparently is fixed now. After updating to 1.5.2-alpha that error is gone but the addresses still do not load. I do not see any errors in the browsers console, all network requests seem to succeed, I do not see any corresponding errors in the server side log files. Browsing through the ajax requests done by the app I can see the address books getting requested, I do NOT see any request to entries in those address books though. But I am not familiar with the details of that process, so I might miss something here. 

Reverting the contacts app to the previously used version 1.4.0.0 brings back all contacts without any issue. 

### Steps to reproduce
1. have an owncloud-9.1.2 system with working contabs app in version 1.4.0.0 and configured addressbooks holding entries. 
2. replace the contacts app by version 1.5.2-alpha as offered in mentioned PR
3. remove the js assets file if necessary
4. load the contacts app

### Expected behaviour
After loading and initializing the app itself and the existing addressbooks the app should load specific addressbook entries and then disable the busy spinner. 

### Actual behaviour
The app loads, initializes and loads the existing addressbook. Then the busy spinner starts, after a few seconds the background graphics "no contacts in here" is loaded (did so too in working version 1.4.0.0). The background graphics is never removed, the busy spinner never stops, no entries are loaded. 

### Server configuration
**Operating system**:
Linux/X86_64

**Web server:**
Apache http server version 2.4

**Database:**
MySQL RDBMS version 5.7

**PHP version:**
PHP 7.0.8 loaded as module into the http server

**ownCloud version:** (see ownCloud admin page)
OC-9.1.2

**Contacts version:** (see ownCloud apps page)
1.4.0.0: working fine
1.5.1: bug as fixed by mentioned PR: client side JS error
1.5.2-alpha as provided by mentioned PR: no error, no entries shown

**Updated from an older ownCloud or fresh install:**
Update by manually replacing script files, installation works without any other issues

**Signing status (ownCloud 9.0 and above):**
???

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

Result: "No errors have been found."
```

**List of activated apps:**
activity
apporder
audioplayer
bookmarks
calendar
comments
contacts
dav
documents
external
federatedfilesharing
federation
files
files_external
files_pdfviewer
files_sharing
files_texteditor
files_trashbin
files_versions
files_videoplayer
firstrunwizard
gallery
impersonate
imprint
mail
notifications
ownnote
piwik
provisioning_api
spreedme
systemtags
tasks
templateeditor
updatenotification

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder

Enabled:
  - activity: 2.3.2
  - apporder: 0.3.2
  - audioplayer: 1.3.1
  - bookmarks: true
  - calendar: 1.4.1
  - comments: 0.3.0
  - contacts: 1.5.1
  - dav: 0.2.7
  - documents: true
  - external: 1.2
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.6.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - impersonate: 0.0.1
  - imprint: 0.2.5
  - mail: 0.6.0
  - notifications: 0.3.0
  - ownnote: 1.08
  - piwik: 0.1.0
  - provisioning_api: 0.5.0
  - spreedme: 0.3.4
  - systemtags: 0.3.0
  - tasks: 0.9.3
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - files_antivirus
  - music
  - news
  - notes
  - spreedme.orig
  - user_external
  - user_ldap

```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)


{
    "system": {
        "debug": false,
        "allow_user_to_change_display_name": true,
        "appcodechecker": false,
        "appstoreenabled": true,
        "appstoreurl": "https:\/\/api.owncloud.com\/v1",
        "appstore.experimental.enabled": false,
        "apps_paths": [
            {
                "path": "\/var\/www\/hosts\/cloud\/public\/apps",
                "url": "\/apps",
                "writable": true
            }
        ],
        "asset-pipeline.enabled": true,
        "assetdirectory": "\/var\/www\/hosts\/cloud\/assets",
        "check_for_working_webdav": true,
        "check_for_working_wellknown_setup": true,
        "check_for_working_htaccess": true,
        "cron_log": false,
        "datadirectory": "\/var\/www\/hosts\/cloud\/data",
        "dbtype": "mysql",
        "version": "9.1.2.5",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "enable_avatars": true,
        "enable_previews": true,
        "filesystem_check_changes": 0,
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "has_internet_connection": true,
        "hashingCost": 10,
        "knowledgebaseenabled": true,
        "loglevel": 2,
        "maintenance": false,
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "forcessl": true,
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",           
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "mail_smtpmode": "php",
        "mail_from_address": "cloud",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "30, auto",
        "updatechecker": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0
        },
        "htaccess.RewriteBase": "\/"
    }
}

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chromium 50.0...
Firefox 49.0.2 (note: spinner is visible, but does not turn any more...)

**Operating system:**
Linux x86_64

**CardDAV-clients:**
n/a

#### Web server error log
```
no relevant entries (related)
```

#### ownCloud log (data/owncloud.log)
```
no relevant entries (related)
```

#### Browser log
```
Insert your browser log here, this could for example include:

no relevant entries (related)
```

**Comment by DeepDiver1975** [comment]
Please open the network console of the browser and look for the REPORT request which is asking for all the contacts - is the response status 207?

like this?

![bildschirmfoto von 2016-12-07 12-52-02](https://cloud.githubusercontent.com/assets/1005065/20966742/156c7394-bc7c-11e6-94ae-5d4fa2e628e0.png)

**Comment by arkascha** [comment]
@DeepDiver1975 
Correct, I see one such REPORT request for each address book, they all show status 207 and they also actually contain the address book entries in their response which I did not notice before. They are not visualized though...

**Comment by DeepDiver1975** [comment]
okay - so - I guess there is at least one vcard which is causing the js code to fail.
Nothing in the error console?

**Comment by arkascha** [comment]
Hm, I played around a bit further: 

Seems there is some unpredictable behavior involved: 
when trying a few times I sometimes do get an error in the console: 
`Uncaught (in promise) Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!(…)`

My stomach says that this is an issue with a partial update of the app... Sorry that I have to ask, but as you know I stopped working with the owncloud code quite some time ago changing to other areas. So could you please give me a hint how I can make sure that really no traces of the older app version are still somehow used? I did clear and disable the browser cache, I did remove the assets js file, I also tried with the debug flag set to true. Anything else I should do?

**Comment by DeepDiver1975** [comment]
get the 1.5.2-alpha package - https://github.com/owncloud/contacts/releases/download/1.5.2-alpha/contacts.tar.gz and extract it fully to apps/contacts

in order to get the browser in a clean state: use a private/incognito browser window :wink:

**Comment by DeepDiver1975** [comment]
> Uncaught (in promise) Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!(…)

can I get the full print out from the console? it can give some indication - THX a lot!

**Comment by arkascha** [comment]
I had to disable the asset pipeline to get something meaningful: 

`angular.js?v=adc76e2…:13920 Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: [[{"msg":"ctrl.contactList","newVal":[],"oldVal":[]}],[{"msg":"ctrl.contactList","newVal":[],"oldVal":"..."}],[{"msg":"ctrl.contactList","newVal":[],"oldVal":"..."}],[{"msg":"ctrl.contactList","newVal":[],"oldVal":"..."}],[{"msg":"ctrl.contactList","newVal":[],"oldVal":"..."}]]
http://errors.angularjs.org/1.5.8/$rootScope/infdig?p0=10&p1=%5B%5B%7B%22ms…contactList%22%2C%22newVal%22%3A%5B%5D%2C%22oldVal%22%3A%22...%22%7D%5D%5D
    at angular.js?v=adc76e2…:68
    at Scope.$digest (angular.js?v=adc76e2…:17562)
    at Scope.$apply (angular.js?v=adc76e2…:17790)
    at contactList_controller.js:75(anonymous function) @ angular.js?v=adc76e2…:13920(anonymous function) @ angular.js?v=adc76e2…:10467Scope.$apply @ angular.js?v=adc76e2…:17792(anonymous function) @ contactList_controller.js:75
angular.js:68 Uncaught (in promise) Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!(…)(anonymous function) @ angular.js?v=adc76e2…:68Scope.$digest @ angular.js?v=adc76e2…:17562Scope.$apply @ angular.js?v=adc76e2…:17790(anonymous function) @ contactList_controller.js:75`

And another hint (maybe): 
As said before the behavior appears to be somewhat non-deterministric. Usually I do not get an error in the console, but if trying often enough then suddenly the errors start to appear. And from then on that seems to be deterministic, there always seems to appear that error, though a different number from time to time. Sorry for that mess!

**Comment by DeepDiver1975** [comment]
I have seen this once during development ... only once ...

Which language is selected for this user?

**Comment by arkascha** [comment]
English language.

**Comment by LeOS-GSI** [comment]
On my 9.1.2 server the connects are still not shown. Is there a way to downgrade ?

**Comment by arkascha** [comment]
Could this be connected to or identical with https://github.com/owncloud/contacts/issues/478?

**Comment by LeOS-GSI** [comment]
Thx, it could, but I don't think so, because I only have around 750 contacts and the vcf file is under 1mb. I have downgraded and now, all contacts are shown again.

**Comment by arkascha** [comment]
@harvey186 Ok, looks like you have the same issue as I do...
Can you spot anything of interest in your log files / the browser console? @DeepDiver1975 certainly is grateful for all hints on this...

**Comment by IwishIcanFLighT** [comment]
I have the same issue on my 9.1.2 OC instance. Broken since 1.4.0, 1.5.1 didn't fix it.
I tried the 1.5.2 alpha this morning and am still facing the same issue. Blank page, nothing shows up beside the parameters gear button (which does nothing).

Firefox console logs:
```
JQMIGRATE: Migrate is installed, version 1.4.0  jquery-migrate.min.js:2:542
SyntaxError: unterminated string literal[En savoir plus]  angular.js:310:9
TypeError: angular is undefined[En savoir plus]  angular-route.js:3:30
ReferenceError: angular is not defined[En savoir plus]  angular-uuid4.js:12:1
TypeError: invalid 'in' operand angular  angular-cache.js:1:339
TypeError: angular is undefined[En savoir plus]  angular-sanitize.js:1:44
ReferenceError: angular is not defined[En savoir plus]  select.js:2:279
ReferenceError: angular is not defined[En savoir plus]  script.js:1:1
ReferenceError: angular is not defined[En savoir plus]  ui-bootstrap.min.js:7:4
ReferenceError: angular is not defined[En savoir plus]  ui-bootstrap-tpls.min.js:7:4
TypeError: a is undefined[En savoir plus]
« window.controllers » est obsolète. Ne l’utilisez pas pour détecter l’agent utilisateur (UA).
```

**Comment by Ascendor** [comment]
I have the same problem. I tried a bit more, and this version: https://github.com/owncloud/contacts/releases/tag/v0.0.0.216 is the last one working fine for me. Releases v0.0.0.217 through v0.0.0.244 have NOT been tested by me as they don't provide a download package.

**Comment by arkascha** [comment]
The issue is still present in OC-9.1.3, all versions above 1.4.0.0 fail to show any contancts.

**Comment by Linkinsoldier** [comment]
Same issue here with NC 10.0.2. and Contacts 1.5.1.

After adding a contact I could not get the app to show me any contacts anymore (Firefox and Chrome). Sync with Davdroid still worked... after deleting about 10 contacts on my cell phone everything works again :)

**Comment by arkascha** [comment]
@Linkinsoldier And were you able to add those 10 contacts again and still have a usable app? Or did it break again which would indicate some limit of the number of contacts that can exist?

**Comment by Linkinsoldier** [comment]
I didnt dare to try readding them. But my contact count is < 200, so i hardly doubt the number is the reason - I think it is the size (around 10MB) as mentioned above!

**Comment by arkascha** [comment]
I can confirm that https://github.com/owncloud/contacts/releases/tag/v0.0.0.216 works for me to (version 1.5.0). 
Version 1.5.1 is broken as is 1.5.2-alpha. 

@DeepDiver1975  Any news on this? It appears more users are affected...

**Comment by wdehoog** [comment]
We have the same problem which is also 'solved' by installing the v.0.0.0.216 version of the contacs app.

We migrated from 8 to 9 and then the problem started. The javascrpt is in an endless loop and when paused stops at line 169 from apps.js. There is also an error 'Error: key must be a string!' at angular-cache.js 491:17 but I do not know if that is related.

**Comment by blitzdesigner** [comment]
I second that, with 1.5.0. is working for me too, evrything above it not.

**Comment by LeOS-GSI** [comment]
I have just tried the update to 1.5.1 and it's still not working. So I will g back again

**Comment by dercorn** [comment]
@DeepDiver1975 @arkascha I can confirm that [v0.0.0.216](https://github.com/owncloud/contacts/releases/tag/v0.0.0.216) works for me, too. The issue came up after an upgrade from 9.1.0 to 9.1.3.

**Comment by Ascendor** [comment]
Has anyone tried it on OC 9.1.4 already? Don't want to go through the hassle of upgrading and downgrading the contacts app again in case anyone else tried it.

**Comment by senshes** [comment]
@Ascendor After Upgrade to OC 9.1.4 no change, contacts 1.5.1 is going to be installed and it is still broken.

**Comment by DeepDiver1975** [comment]
@senshes - mind testing the current development stage? 
[contacts.zip](https://github.com/owncloud/contacts/files/750665/contacts.zip)

**Comment by senshes** [comment]
@DeepDiver1975  installed, tested, works like a charm on my server, no issues found - thanks a lot

**Comment by arkascha** [comment]
@DeepDiver1975 It appears this version you linked here 10 days ago is a step forward: 
Contacts do get loaded for me too now. Yeah! Thanks! 

However there is a new issue in that version now: 
Some of the contacts (about half of mine) show a broken display in the list of contacts: 
there name is shown as ["Name","Surname"], with the literal square brackets around the name and the icon to the left of the name shows a "{{" instead of the first letter of the name...

**Comment by DeepDiver1975** [comment]
> However there is a new issue in that version now:
> Some of the contacts (about half of mine) show a broken display in the list of contacts:
> there name is shown as ["Name","Surname"], with the literal square brackets around the name and the icon to the left of the name shows a "{{" instead of the first letter of the name...

mind opening a new issue with an example vcard attached? I'll happily debug into this then .... THX

**Comment by DeepDiver1975** [comment]
> Contacts do get loaded for me too now. Yeah! Thanks!

Nice! Let me release this version right after your other bug is fixed

**Comment by arkascha** [comment]
@DeepDiver1975 Create issue #579

**Comment by Linkinsoldier** [comment]
May I ask if this bugfix made it into 1.5.3 of the app? I have 2 contact addressbooks near 10MB and would be grateful for this fix :)

**Comment by DeepDiver1975** [comment]
1.5.3 was just published - please give it a try - thx

I'll close this issue - please open a new new for 1.5.3 - THX
